lib/src/helpers/lists.dart

```diff
@@ -5,40 +5,41 @@
 import 'dart:collection';
 import 'dart:js_interop';
 
-/// Defines interop members on a JSObject that would be present on a js list object.
-/// The JSObject is assumed to be one of the js list type as right now
-/// we don't have a single interface that would represent all the js list types.
-extension on JSObject {
+/// Defines interop members on a JSObject that would be present on a JS list object.
+/// The JSObject is assumed to be one of the JS list type as right now
+/// we don't have a single interface that would represent all the JS list types.
+extension type _JSList<T extends JSObject>(JSObject _) implements JSObject {
   /// The **`item()`** method returns the [JSObject]
   /// at the specified index in the list.
-  external JSObject item(int index);
+  external T item(int index);
 
   /// The **`length`** read-only property indicates the number of
   /// items in a given list.
   external int get length;
 }
 
+/// A wrapper to present a JS immutable list of type T as a `List<U>` where U
+/// is the list item type.
+class JSImmutableListWrapper<T extends JSObject, U extends JSObject>
+    extends Object with ListMixin<U> implements List<U> {
+  final T _original;
+  JSImmutableListWrapper(this._original);
 
-/// A wrapper to present a js list as a `List<T>`.
-class JSListWrapper<T extends JSObject> extends Object
-    with ListMixin<T>
-    implements List<T> {
-  final JSObject _original;
-  JSListWrapper(this._original);
+  _JSList<U> get _jsList => _JSList<U>(_original);
```

Comment:
nit: A production compiler will be smart enough to inline these calls, but DDC won't. We could make `_jsList` a `final` (possibly `late` if needed) variable instead of a getter to avoid the indirection.

Reply:
Yup, that's a good point. An alternative is to only keep `_jsList` since we don't expose `_original` yet (maybe in a future CL as I do think that's handy), but it's okay to leave it as is too.

> Also, could you clarify on this? Curious why that would be the case with this getter?

DDC for the most part just directly emits the equivalent JS code, so it'll create a separate member for the getter requiring a method indirection. In practice, v8 and other browser engines are probably smart enough that this doesn't make a functional difference.

Reply:
> Yup, that's a good point. An alternative is to only keep `_jsList` since we don't expose `_original` yet (maybe in a future CL as I do think that's handy), but it's okay to leave it as is too.

Cool! I'll keep only `_jsList` for now just to be on the safer side but also doing things like in above code upsets the analyzer abt unused value of `_original`.